kong/types.go

```diff
@@ -267,3 +270,84 @@ type Role struct {
 	Comment   *string `json:"comment,omitempty" yaml:"comment,omitempty"`
 	IsDefault *bool   `json:"is_default,omitempty" yaml:"is_default,omitempty"`
 }
+
+// PermissionRoleID represents a Role identifier within a permission struct
+type PermissionRoleID struct {
+	ID *string `json:"id,omitempty" yaml:"id,omitempty"`
+}
+
+// EndpointPermission represents an RBAC Endpoint Permisson in Kong Enterprise
+type EndpointPermission struct {
+	CreatedAt *int              `json:"created_at,omitempty" yaml:"created_at,omitempty"`
+	Workspace *string           `json:"workspace,omitempty" yaml:"workspace,omitempty"`
+	Endpoint  *string           `json:"endpoint,omitempty" yaml:"endpoint,omitempty"`
+	Actions   []*string         `json:"actions,omitempty" yaml:"actions,omitempty"`
+	Negative  *bool             `json:"negative,omitempty" yaml:"negative,omitempty"`
+	Role      *PermissionRoleID `json:"role,omitempty" yaml:"role,omitempty"`
```

Comment:
Question: Why can you reference the `RBACRole` struct here? That is how we do foreign-key referencing with other related entities.